main.gs

```diff
@@ -31,3 +31,44 @@ function matchGirls() {
         outputSheet.appendRow([firstName, lastName, preference_1, preference_2, preference_3]);
     }
 }
+
+function scorer() {
+    var score = 0;
+    var responseSheet = SpreadsheetApp.getActiveSheet();
+    var responseData = responseSheet.getDataRange().getValues();
+    
+    var outputSheet = SpreadsheetApp.openById(OUTPUT_SHEET_ID);
+    var outputData = outputSheet.getDataRange().getValues();
+
+    for (var i = 1; i < responseData.length; i++) {
+        if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_1])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_1])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_1])) {
+            score += 1;
+        }
+        if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_2])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_2])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_2])) {
+            score += 3;
+        }
+        if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_3])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_3])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_3])) {
+            score += 5;
+        }
+        if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_4])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_4])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_4])) {
+            score += 10;
+        }if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_5])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_5])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_5])) {
+            score += 11;
+        }if((outputData[i][2] == responseData[i][COLUMN_PREFERENCE_6])
+            || (outputData[i][3] == responseData[i][COLUMN_PREFERENCE_6])
+            || (outputData[i][4] == responseData[i][COLUMN_PREFERENCE_6])) {
+            score += 12;
```

Comment:
Write constants to remove these magic numbers. Can you refactor this block to reduce repetition? What about a helper function?